src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java

```diff
@@ -0,0 +1,97 @@
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.storage.api.FeedbackResponseCommentsDb;
+
+/**
+ * SUT: {@link FeedbackResponseCommentsDb},
+ *      {@link teammates.storage.search.FeedbackResponseCommentDocument},
+ *      {@link teammates.storage.search.FeedbackResponseCommentSearchQuery}.
+ */
+public class FeedbackResponseCommentSearchTest extends BaseSearchTest {
+
+    @Test
+    public void allTests() {
+        FeedbackResponseCommentsDb commentsDb = new FeedbackResponseCommentsDb();
+
+        FeedbackResponseCommentAttributes frc1I1Q1S1 = dataBundle.feedbackResponseComments
+                .get("comment1FromT1C1ToR1Q1S1C1");
+        FeedbackResponseCommentAttributes frc1I1Q2S1 = dataBundle.feedbackResponseComments
+                .get("comment1FromT1C1ToR1Q2S1C1");
+        ArrayList<InstructorAttributes> instructors = new ArrayList<InstructorAttributes>();
+        instructors.add(dataBundle.instructors.get("instructor1OfCourse1"));
+
+        ______TS("success: search for comments in whole system; query string does not match any comment");
+
+        FeedbackResponseCommentSearchResultBundle bundle = commentsDb.search("non-existent", instructors);
+        assertEquals(0, bundle.numberOfResults);
+        assertTrue(bundle.comments.isEmpty());
+
+        ______TS("success: search for comments in whole system; query string matches single comment");
+
+        bundle = commentsDb.search("\"Instructor 1 comment to student 1 self feedback Question 2\"", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1);
+
+        ______TS("success: search for comments in instructor's course; query string matches some comments");
+
+        bundle = commentsDb.search("\"self feedback\"", instructors);
+        verifySearchResults(bundle, frc1I1Q1S1, frc1I1Q2S1);
+
+        ______TS("success: search for comments in instructor's course; query string is case in-sensitive");
+
+        bundle = commentsDb.search("\"Instructor 1 COMMENT to student 1 self feedback Question 2\"", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1);
+
+        ______TS("success: search for comments searchable by feedbackSessionName");
+
+        bundle = commentsDb.search("\"First feedback session\"", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1, frc1I1Q1S1);
+
+        ______TS("success: search for comments using Instructor's email");
+
+        bundle = commentsDb.search("instructor1@course1.tmt", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1, frc1I1Q1S1);
+
+        ______TS("success: search for comments using Student name");
+
+        bundle = commentsDb.search("Student in two courses", instructors);
+        verifySearchResults(bundle, frc1I1Q1S1, frc1I1Q2S1);
+
+        ______TS("success: search for comments; no results for deleted comment");
```

Comment:
`no results for deleted comment` -> `confirms deleted comments are not included in results`